xwiki-platform-core/xwiki-platform-localization/xwiki-platform-localization-script/src/main/java/org/xwiki/localization/script/LocalizationScriptService.java

```diff
@@ -267,4 +269,15 @@ public String render(String key, Syntax syntax, Collection<?> parameters, Locale
 
         return result;
     }
+
+    /**
+     * @return the list of all known locales
+     * @since 9.7RC1
+     * @since 8.4.6
+     * @since 9.6.1
+     */
+    public ULocale[] getKnownLocales()
+    {
+        return ULocale.getAvailableLocales();
```

Comment:
Using the ICU API somewhere only make sense if you use it everywhere. For the bridge, see http://userguide.icu-project.org/icu4j-locale-service-provider.